scene/gui/text_edit.cpp

```diff
@@ -6159,16 +6183,64 @@ void TextEdit::set_code_hint(const String &p_hint) {
 	update();
 }
 
-void TextEdit::code_complete(const Vector<String> &p_strings, bool p_forced) {
+void TextEdit::code_complete(const List<ScriptCodeCompletionOption> &p_strings, bool p_forced) {
 
-	completion_strings = p_strings;
+	completion_sources = p_strings;
 	completion_active = true;
 	completion_forced = p_forced;
-	completion_current = "";
+	completion_current = ScriptCodeCompletionOption();
 	completion_index = 0;
 	_update_completion_candidates();
 }
 
+Ref<Texture> TextEdit::get_completion_icon(const ScriptCodeCompletionOption &p_option) {
+	Ref<Texture> tex = NULL;
+#ifdef TOOLS_ENABLED
```

Comment:
Can the icon not be a value in the `ScriptCodeCompletionOption` struct?

Reply:
I don't think it is good to put icons in `ScriptCodeCompletionOption` as the icons are inferred from the `kind`. If so we have to set icons every where we make the options.

Reply:
Could it not be set as part of the `_complete_request` [here?](https://github.com/godotengine/godot/pull/29744/files#diff-86e49621fcb98c850c78e73214c8b478L714)

Reply:
That is a good idea